cluster/charts/crossplane/templates/deployment.yaml

```diff
@@ -116,6 +149,16 @@ spec:
             - key: {{ .Values.registryCaBundleConfig.key }}
               path: {{ .Values.registryCaBundleConfig.key }}
       {{- end }}
+      {{- if .Values.webhooks.enabled }}
+      - name: webhook-tls-secret
+        secret:
+          secretName: webhook-tls-secret
+          items:
```

Comment:
I am not a 100% sure with the Let's Encrypt case since that certificate would be signed by an already trusted root CA and kube-apiserver may not need an additional cabundle in that case. But, I also suspect that it would not possible at all to generate that certificate with Let's Encrypt since it would attempt to [verify](https://cert-manager.io/docs/configuration/acme/#solving-challenges) that you own the domain.

The requirement about being self-signed is coming from the fact that we are passing it `tls.crt` as `cabundle` which means it needs to be a (root or intermediate) CA in addition to being the webhook server certificate. I was thinking that this would mean it to be a self-signed certificate unless it is being signed by an already trusted CA (e.g. root ca of kubernetes API itself).

@ulucinar please correct me if I am wrong / missing something :) ?